phoenix-core-client/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -1530,15 +1531,23 @@ public static void moveHBaseLevelTTLToSYSCAT(PhoenixConnection oldMetaConnection
                             byte[] tableName = rowKeyMetaData[PhoenixDatabaseMetaData.TABLE_NAME_INDEX];
 
                             String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+                            currentTableName = fullTableName;
                             if (SchemaUtil.isSystemTable(SchemaUtil.getTableNameAsBytes(schemaName, tableName))) {
                                 //We do not support system table ttl through phoenix ttl, and it will be moved to a
                                 //constant value in future commit.
                                 continue;
                             }
+                            PTable pTable = oldMetaConnection.getTableFromServerNoCache(schemaName, tableName);
```

Comment:
As all we want is emptyCF, we don't need to make syscat call if PTable is already in cache? So, we can use `oldMetaConnection.getTable(null, fullTableName)` instead?